mos-platform/cx16/cx16.h

```diff
@@ -0,0 +1,325 @@
+/*                                                                           */
+/*****************************************************************************/
+
+#ifndef _CX16_H
+#define _CX16_H
+
+/* Check for errors */
+#ifndef __CX16__
+#  error This module may be used only when compiling for the CX16!
+#endif
+
+
+
+/*****************************************************************************/
+/*                                   Data                                    */
+/*****************************************************************************/
+
+
+
+/* Additional output character codes */
+#define CH_COLOR_SWAP           0x01
+#define CH_UNDERLINE            0x04
+#define CH_WHITE                0x05
+#define CH_BOLD                 0x06
+#define CH_BACKSPACE            0x08
+#define CH_ITALIC               0x0B
+#define CH_OUTLINE              0x0C
+#define CH_FONT_ISO             0x0F
+#define CH_RED                  0x1C
+#define CH_GREEN                0x1E
+#define CH_BLUE                 0x1F
+#define CH_ORANGE               0x81
+#define CH_FONT_PET             0x8F
+#define CH_BLACK                0x90
+#define CH_ATTR_CLEAR           0x92
+#define CH_BROWN                0x95
+#define CH_PINK                 0x96
+#define CH_LIGHTRED             CH_PINK
+#define CH_GRAY1                0x97
+#define CH_GRAY2                0x98
+#define CH_LIGHTGREEN           0x99
+#define CH_LIGHTBLUE            0x9A
+#define CH_GRAY3                0x9B
+#define CH_PURPLE               0x9C
+#define CH_YELLOW               0x9E
+#define CH_CYAN                 0x9F
+#define CH_SHIFT_SPACE          0xA0
+
+/* Additional key defines */
+#define CH_SHIFT_TAB            0x18
+#define CH_HELP                 0x84
+#define CH_F1                   0x85
+#define CH_F2                   0x89
+#define CH_F3                   0x86
+#define CH_F4                   0x8A
+#define CH_F5                   0x87
+#define CH_F6                   0x8B
+#define CH_F7                   0x88
+#define CH_F8                   0x8C
+#define CH_F9                   0x10
+#define CH_F10                  0x15
+#define CH_F11                  0x16
+#define CH_F12                  0x17
+
+/* Color defines */
+#define COLOR_BLACK             0x00
+#define COLOR_WHITE             0x01
+#define COLOR_RED               0x02
+#define COLOR_CYAN              0x03
+#define COLOR_PURPLE            0x04
+#define COLOR_GREEN             0x05
+#define COLOR_BLUE              0x06
+#define COLOR_YELLOW            0x07
+#define COLOR_ORANGE            0x08
+#define COLOR_BROWN             0x09
+#define COLOR_PINK              0x0A
+#define COLOR_LIGHTRED          COLOR_PINK
+#define COLOR_GRAY1             0x0B
+#define COLOR_GRAY2             0x0C
+#define COLOR_LIGHTGREEN        0x0D
+#define COLOR_LIGHTBLUE         0x0E
+#define COLOR_GRAY3             0x0F
+
+/* TGI color defines */
+#define TGI_COLOR_BLACK         COLOR_BLACK
```

Comment:
I consider TGI bloat (but I'm open to having my mind changed), so I don't really intend to support it. Accordingly, we can remove these until we're actually trying to get TGI working (currently slated for never).

Reply:
TGI removed